plugins/dicom_viewer/server/__init__.py

```diff
@@ -1,59 +1,117 @@
+        Description('Get and store common DICOM metadata, if any, for all files in the item.')
+        .modelParam('id', 'The item ID',
+                    model='item', level=AccessType.WRITE, paramType='path')
         .errorResponse('ID was invalid.')
         .errorResponse('Read permission denied on the item.', 403)
     )
-    def getDicom(self, item, filters, force):
-        if force:
-            Item().requireAccess(item, user=self.getCurrentUser(), level=AccessType.WRITE)
-        filters = set(filter(None, filters.split(',')))
-        files = list(Item().childFiles(item))
-        # process files if they haven't been processed yet
-        for i, f in enumerate(files):
-            if force or 'dicom' not in f:
-                files[i] = process_file(f)
-        # filter out non-dicom files
-        files = [x for x in files if x.get('dicom')]
-        # sort files
-        files = sorted(files, key=sort_key)
-        # execute filters
-        if filters:
-            for f in files:
-                dicom = f['dicom']
-                dicom = dict((k, dicom[k]) for k in filters if k in dicom)
-                f['dicom'] = dicom
-        return files
-
-
-def sort_key(f):
-    dicom = f.get('dicom', {})
+    def makeDicomItem(self, item):
+        """Try to convert an existing item into a "DICOM item", which contains a
+        "dicomMeta" field with DICOM metadata that is common to all DICOM files.
+        """
+        metadataReference = None
+        dicomFiles = []
+
+        for file in Item().childFiles(item):
+            dicomMeta = parse_file(file)
+            if dicomMeta is None:
+                continue
+            dicomFiles.append(_extractFileData(file, dicomMeta))
+            if metadataReference is None:
+                metadataReference = dicomMeta
+            else:
+                metadataReference = removeUniqueMetadata(metadataReference, dicomMeta)
+
+        if len(dicomFiles) is not 0:
+            # Sort the dicom files
+            dicomFiles = sorted(dicomFiles, key=_sort_key)
+            # Store in the item
+            item['dicom'] = {}
+            item['dicom']['meta'] = metadataReference
+            item['dicom']['files'] = dicomFiles
+            # Save the item
+            Item().save(item)
+
+
+def _extractFileData(file, dicom):
+    """Extract the usefull data to be stored in the `item['dicom']['files']`.
+    In this way it become simpler to sort them and store them.
+    """
+    return {
+        '_id': file.get('_id'),
+        'name': file.get('name'),
+        'SeriesNumber': dicom.get('SeriesNumber'),
+        'InstanceNumber': dicom.get('InstanceNumber'),
+        'SliceLocation': dicom.get('SliceLocation')
+    }
+
+
+def _sort_key(f):
+    """These properties are used to sort the files into the item."""
     return (
-        dicom.get('SeriesNumber'),
-        dicom.get('InstanceNumber'),
-        dicom.get('SliceLocation'),
-        f['name'],
+        f.get('name'),
```

Comment:
I think `name` should be a final fallback, if the first 3 fields fail to disambiguate slices.